File: GameLibrary/src/cc/lib/reflector/ParseException.java
package cc.lib.reflector;

import java.io.IOException;

public class ParseException extends IOException {
    final int lineNum;

    ParseException(int lineNum, String msg) {
        super(msg);
        this.lineNum = lineNum;
    }

    ParseException(int lineNum, Exception e) {
        super(e);
        this.lineNum = lineNum;
    }

    @Override
    public String getMessage() {
        return "Line (" + lineNum + ") " + super.getMessage();
    }
}

File: GameLibrary/src/cc/lib/kreflector/ParseException.kt
package cc.lib.kreflector

import java.io.IOException

class ParseException : IOException {
	val lineNum: Int

	internal constructor(lineNum: Int, msg: String?) : super(msg) {
		this.lineNum = lineNum
	}

	internal constructor(lineNum: Int, e: Exception?) : super(e) {
		this.lineNum = lineNum
	}

	override val message: String
		get() = "Line (" + lineNum + ") " + super.message
}